nativelib/src/main/resources/scala-native/gc/none/gc.c

```diff
@@ -1,30 +1,83 @@
 #include <stdlib.h>
 #include <sys/mman.h>
-
+#include "Parsing.h"
+#include "../commix/Memory.h"
 // Darwin defines MAP_ANON instead of MAP_ANONYMOUS
 #if !defined(MAP_ANONYMOUS) && defined(MAP_ANON)
 #define MAP_ANONYMOUS MAP_ANON
 #endif
 
 // Dummy GC that maps chunks of 4GB and allocates but never frees.
 
-// Map 4GB
-#define CHUNK (4 * 1024 * 1024 * 1024L)
 // Allow read and write
 #define DUMMY_GC_PROT (PROT_READ | PROT_WRITE)
 // Map private anonymous memory, and prevent from reserving swap
-#define DUMMY_GC_FLAGS (MAP_NORESERVE | MAP_PRIVATE | MAP_ANONYMOUS)
+#define DEFAULT_DUMMY_GC_FLAGS (MAP_NORESERVE | MAP_PRIVATE | MAP_ANONYMOUS)
 // Map anonymous memory (not a file)
 #define DUMMY_GC_FD -1
 #define DUMMY_GC_FD_OFFSET 0
 
 void *current = 0;
 void *end = 0;
 
+static size_t DEFAULT_CHUNK;
+static size_t PREALLOC_CHUNK;
+static size_t CHUNK;
+static size_t DUMMY_GC_FLAGS = DEFAULT_DUMMY_GC_FLAGS;
+static size_t TO_NORMAL_MMAP = 1L;
+
+size_t set_prealloc() {
+#ifdef __linux__
+    return (MAP_NORESERVE | MAP_PRIVATE | MAP_ANONYMOUS | MAP_POPULATE);
+#else
+    // MAP_POPULATE is linux exclusive
+    return DEFAULT_DUMMY_GC_FLAGS(MAP_NORESERVE | MAP_PRIVATE | MAP_ANONYMOUS);
```

Comment:
Is it correct? Should it not be just `DEFAULT_DUMMY_GC_FLAGS`? 
Also, we might think about removing the `DUMMY_` part from all names, since in most cases those are actual/final values.